packages/next/build/webpack/loaders/next-serverless-loader/page-handler.ts

```diff
@@ -347,7 +347,7 @@ export function getPageHandler(ctx: ServerlessHandlerCtx) {
                 poweredByHeader,
               },
               {
-                private: isPreviewMode,
+                private: isPreviewMode || renderOpts.isNotFound,
```

Comment:
We need `cache-control` to be set when returning `notFound: true` or it won't cache the 404 in that case as expected and wait to revalidate in the proxy. 

It looks like we want this to be private specifically for the 404 page with `getStaticProps` and not when using `notFound: true` which we can tell by using the `is404Page` check instead